src/mito2/src/manifest/manager.rs

```diff
@@ -275,6 +275,126 @@ impl RegionManifestManager {
         self.stopped = true;
     }
 
+    /// Installs the manifest changes from the current version to the target version.
+    ///
+    /// Returns installed version.
+    /// **Note**: This function is not guaranteed to install the target version strictly.
+    /// The installed version may be greater than the target version.
+    pub async fn install_manifest_to(
+        &mut self,
+        target_version: ManifestVersion,
+    ) -> Result<ManifestVersion> {
+        let _t = MANIFEST_OP_ELAPSED
+            .with_label_values(&["install_manifest_to"])
+            .start_timer();
+
+        // Case 1: If the target version is less than the current version, return the current version.
+        if self.last_version >= target_version {
+            debug!(
+                "Target version {} is less than or equal to the current version {}, region: {}, skip install",
+                target_version, self.last_version, self.manifest.metadata.region_id
+            );
+            return Ok(self.last_version);
+        }
+
+        ensure!(
+            !self.stopped,
+            RegionStoppedSnafu {
+                region_id: self.manifest.metadata.region_id,
+            }
+        );
+
+        // Fetches manifests from the last version strictly.
+        let mut manifests = self
+            .store
+            .fetch_manifests_strict_from(self.last_version, target_version + 1)
+            .await?;
```

Comment:
Can we return the manifests and the checkpoints in the range? 

If the checkpoint exists, we can install the checkpoint and avoid fetching the checkpoint again.